### PR TITLE
Enable TDX attestation 

### PR DESCRIPTION
**Pull request by bprashan**
- Enabled TDX attestation
- refactor automation scripts

### CODE REVIEW

**Comment by bprashan** [comment]
verified Jenkins configuration changes and automation script changes on development jenkins job.
http://ba5sapp0350:8080/job/bprashan_tdx_1/60/

File: ci/tdx_canonical_linux_stack.sh [hunk 1]
```diff
@@ -25,168 +33,24 @@ if [[ $DISTRO_VER == "23.10" ]]; then
...
-        if [ $? -ne 0 ]; then
-                exit 1
-        fi
-}
-
-setup_canonical_suite(){
-	echo "Installing Canonical suite required packages on host system"
-	sudo apt install tox python3
-}
+if [[ -z "$JENKINS_URL" ]]; then
```
Comment (jinengandhi-intel) [inline ci/tdx_canonical_linux_stack.sh:36]: What's the intention behind this check?

File: ci/utils/canonical/config.json
Comment (jinengandhi-intel) [inline ci/utils/canonical/config.json:1]: rename the canonical folder to ubuntu. Ubuntu is the distro, Canonical is a company that supports commercial Ubuntu users.

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

File: ci/utils/canonical/run_automated_tests.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                ./run.sh -g ubuntu -s all
+        fi
+        if [ $? -ne 0 ]; then
+                exit 1
+        fi
+}
+
+setup_canonical_suite(){
+        echo "Installing Canonical suite required packages on host system"
+        sudo apt install tox python3
```
Comment (anjalirai-intel) [inline ci/utils/canonical/run_automated_tests.sh]: It should be `apt install -y tox python3`

File: ci/utils/canonical/run_automated_tests.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+run_canonical_suite(){
+        cd $TDX_DIR/tests/tests
+        export TDXTEST_GUEST_IMG=$QCOW2_IMG
+        sudo -E tox -e test_specify -- "not tdreport and not perf_benchmark"
```
Comment (anjalirai-intel) [inline ci/utils/canonical/run_automated_tests.sh]: sudo is not required

File: ci/tdx_canonical_linux_stack.sh (hunk 1 above)
Comment (bprashan) [reply ci/tdx_canonical_linux_stack.sh:36]: To ensure that shell script is invoke via jenkins pipeline

File: ci/utils/canonical/config.json
Comment (bprashan) [reply ci/utils/canonical/config.json:1]: renamed to ubuntu

File: ci/utils/canonical/run_automated_tests.sh (hunk 2 above)
Comment (bprashan) [reply ci/utils/canonical/run_automated_tests.sh]: removed `sudo`

File: ci/utils/canonical/run_automated_tests.sh (hunk 3 above)
Comment (bprashan) [reply ci/utils/canonical/run_automated_tests.sh]: removed `sudo`

**Review by jinengandhi-intel — APPROVED** [review]

**Review by anjalirai-intel — APPROVED** [review]

**Pull request merged**